### ISSUE TITLE
[FF] scrobler stops on ajax navigation

### ISSUE THREAD

**Issue by OwnageIsMagic** [david-sabata/web-scrobbler#1455]
https://bugzilla.mozilla.org/show_bug.cgi?id=1406765 marked as `wontfix`, but it's really annoying that I can't browse site while listening to music on the same tab

**Comment by alexesprit** [comment]
As I said before I have no way to fix this issue on the extension side. That's why I filed the issue in the Mozilla bugtracker you mentioned.

**Comment by OwnageIsMagic** [comment]
The only mention of it is [this comment](https://github.com/david-sabata/web-scrobbler/issues/1423#issuecomment-334907055) on unrelated issue, so it's not a duplicate
Regarding the problem we can vote on this bug. Even 5 votes can promote it to the upper quarter of the list.

**Comment by alexesprit** [comment]
I added a workaround here: https://github.com/alexesprit/web-scrobbler/tree/feature/pageaction-fix

Is this acceptable? Can you install the extension as a temporary add-on from this branch and test it?

**Comment by alexesprit** [comment]
@OwnageIsMagic could you check how does v2.2.0 work?

**Comment by OwnageIsMagic** [comment]
Music was playing all of the time
here is screenshot from my lastfm library (https://www.last.fm/user/  /library)
![image](https://user-images.githubusercontent.com/10534634/32819223-0eeacb72-c9d9-11e7-9629-e470085f7fe6.png)
I refreshed page several times to keep it scrobbling
As you may see it scrobbled 2 tracks and just stopped
And I'm sure that there were no tracks longer than 7 minutes

**Comment by alexesprit** [comment]
I noticed the event listener stops receiving track info from the script injected in VK page by unknown reason.

**Comment by alexesprit** [comment]
@OwnageIsMagic @IRainman I added a dirty hack to (probably) resolve this issue. Could you check the test extension from [this](https://drive.google.com/open?id=0B3SDlhZ1oahMT25TU1hQVVVYOEE) zip? I'm testing this one right now and seems like it works w/o any problems.

The problem is the event listener set in `connectors/vk.js` stops working after some time, whereas events from injected scripts are sent correctly.

**Comment by alexesprit** [comment]
Should be fixed in v2.2.2

**Comment by OwnageIsMagic** [comment]
What about browser console?

**Comment by IRainman** [comment]
@alexesprit Firefox version 57.0 is very unstable, there it is broken or buggy almost everything, unfortunately.
P.S. Thank you!

**Comment by OwnageIsMagic** [comment]
Log spam
```
03:51:33.076
Object { artistSelector: null, trackSelector: null, albumSelector: null, currentTimeSelector: null, remainingTimeSelector: null, durationSelector: null, timeInfoSelector: null, artistTrackSelector: null, playButtonSelector: null, playerSelector: null, … }
vk.js:41:2
03:51:33.361
Object { artistSelector: null, trackSelector: null, albumSelector: null, currentTimeSelector: null, remainingTimeSelector: null, durationSelector: null, timeInfoSelector: null, artistTrackSelector: null, playButtonSelector: null, playerSelector: null, … }
vk.js:41:2
03:51:33.534
Object { artistSelector: null, trackSelector: null, albumSelector: null, currentTimeSelector: null, remainingTimeSelector: null, durationSelector: null, timeInfoSelector: null, artistTrackSelector: null, playButtonSelector: null, playerSelector: null, … }
vk.js:41:2
03:51:33.643
Object { artistSelector: null, trackSelector: null, albumSelector: null, currentTimeSelector: null, remainingTimeSelector: null, durationSelector: null, timeInfoSelector: null, artistTrackSelector: null, playButtonSelector: null, playerSelector: null, … }
vk.js:41:2
03:51:33.761
Object { artistSelector: null, trackSelector: null, albumSelector: null, currentTimeSelector: null, remainingTimeSelector: null, durationSelector: null, timeInfoSelector: null, artistTrackSelector: null, playButtonSelector: null, playerSelector: null, … }
```

**Comment by alexesprit** [comment]
Oops. Will publish an update soon.

**Comment by OwnageIsMagic** [comment]
`this.onStateChanged()` is checking `this.isPlaying()` which is always undefined for vk.com
Is it ok?

**Comment by alexesprit** [comment]
@OwnageIsMagic that's not ok.

**Comment by OwnageIsMagic** [comment]
isPlaying checks only for css selector, add additional checks?
But I think you know better what to do

**Comment by alexesprit** [comment]
> isPlaying checks only for css selector, add additional checks?

This is default implementation, other connectors use own implementation (or don't use).

Fixed in add2541046f0df13a2c349033ace78def7f7696a.
